web/src/app/add-stuff/add-stuff.component.spec.ts

```diff
@@ -1,28 +1,90 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule} from '@angular/common/http/testing';
+import {FormsModule, ReactiveFormsModule} from '@angular/forms';
+import {BrowserModule} from '@angular/platform-browser';
+import { RouterTestingModule } from '@angular/router/testing';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import {MatAutocompleteModule, MatChipsModule, MatIconModule, MatInputModule} from '@angular/material';
+import {ConfigService} from '../_services/config.service';
+import {StuffService} from '../_services/stuff.service';
+import {UserService} from '../_services/user.service';
+import { AddCategoryComponent } from '../add-category/add-category.component';
 import { AddStuffComponent } from './add-stuff.component';
-import {FormsModule, ReactiveFormsModule} from "@angular/forms";
-import {BrowserModule} from "@angular/platform-browser";
+import {DebugElement} from '@angular/core';
+import {Observable} from 'rxjs';
+import {By} from '@angular/platform-browser';
 
-
-xdescribe('AddStuffComponent', () => {
+describe('AddStuffComponent', () => {
   let component: AddStuffComponent;
-  let fixture: ComponentFixture<AddStuffComponent>;
+  let fixture: ComponentFixture < AddStuffComponent > ;
+  let stuffService: StuffService;
+
 
-  beforeEach(async(() => {
+  beforeEach(async (() => {
     TestBed.configureTestingModule({
-      imports: [BrowserModule, FormsModule, ReactiveFormsModule],
-      declarations: [ AddStuffComponent ]
-    })
-    .compileComponents();
+        imports: [BrowserModule, FormsModule, ReactiveFormsModule,
+          MatAutocompleteModule, MatChipsModule, MatIconModule, MatInputModule,
+          HttpClientTestingModule, RouterTestingModule, BrowserAnimationsModule
+        ],
+        declarations: [AddStuffComponent, AddCategoryComponent],
+        providers: [StuffService, UserService, ConfigService]
+      })
+      .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(AddStuffComponent);
     component = fixture.componentInstance;
+    stuffService = TestBed.get(StuffService);
     fixture.detectChanges();
   });
+  describe('form component', () => {
+    it('should be created', () => {
+      expect(component).toBeTruthy();
+    });
+    it('should call stuffService on submit', () => {
+      let observable = new Observable();
+      spyOn(observable, 'subscribe');
+      spyOn(stuffService, 'addStuff').and.returnValue(observable);
+      component.name.setValue('stuff name');
+      fixture.detectChanges();
+      component.save();
+
+      expect(stuffService.addStuff).toHaveBeenCalledWith(component.addStuff.value, []);
```

Comment:
It would be good here to fill the form and check that service get it

Reply:
Got it. I'll add it.